projects/igniteui-angular/src/lib/directives/text-selection/text-selection.directive.ts

```diff
@@ -8,6 +8,36 @@ export class IgxTextSelectionDirective {
 
     private selectionState = true;
 
+    /**
+     * Determines whether the input element should be selectable through the directive.
+     *
+     * ```html
+     * <!--set-->
+     * <input
+     *   type="text"
+     *   id="firstName"
+     *   [igxTextSelection]="true">
+     * </input>
+     *
+     * <input
+     *   type="text"
+     *   id="lastName"
+     *   igxTextSelection
+     *   [selected]="true">
+     * </input>
+     * ```
+     *
+     * ```typescript
+     * // get
+     * @ViewChild('firstName',
+     *  {read: IgxTextSelectionDirective})
+     * public firstName: IgxTextSelectionDirective;
+     *
+     * public getFirstNameSelectionStatus() {
+     *  return this.firstName.selected;
+     * }
+     * ```
+     */
     @Input('igxTextSelection')
```

Comment:
For better generation of the documentation it would be more accurate if we could have two separate comments about the getter and the setter accessors.
My proposal for getter accessor: Returns whether the input element is selectable through the directive.
for setter accessor: Determines whether the input element could be selected through the directive.
And we could separate the snippets for the getter and setter respectively.